girder/web_client/src/views/widgets/ItemListWidget.js

```diff
@@ -140,6 +147,110 @@ var ItemListWidget = View.extend({
         this.checked = _.map(this.$('.g-list-checkbox:checked'), function (checkbox) {
             return $(checkbox).attr('g-item-cid');
         }, this);
+    },
+
+    centerSelected: function () {
+        const widgetcontainer = $('.g-hierarchy-widget-container');
+        const selected = $('li.g-item-list-entry.g-selected');
+
+        if (widgetcontainer.length > 0 && selected.length > 0) {
+            const centerPos = (widgetcontainer.height() / 2.0) + (selected.outerHeight() / 2.0);
+            // Take the current scroll top position and add it to the position of the top of the selected item, then center it
+            const scrollPos = widgetcontainer[0].scrollTop + selected.position().top - centerPos;
+            if (this.tempScrollPos === undefined) {
+                this.tempScrollPos = scrollPos;
+            }
+            // Call a custom scroll event to prevent thinking the user initiated it
+            var e = new CustomEvent('scroll', { detail: 9999 });
```

Comment:
Instead of using `9999`, I think this would be more obvious if it were an explicit string, like `browser_widget_scroll_event`, unless there is some reason this needs to be a number.